### PR TITLE
Add new error for EJB Custom Binding elements having a Namespace

### PR DESCRIPTION
**Pull request by olendvcook**
Added `NAMESPACE_IN_JNDI_BINDING_NAME_CNTR0339E` and `NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E`

I had to update the number for `NAME_ALREADY_BOUND_WARN_CNTR0336W=CNTR0336W` to `NAME_ALREADY_BOUND_WARN_CNTR0338W=CNTR0338W`

NOTE: This warning message is currently not used yet, but has already been delivered and translated and was put in for a feature currently in beta, the only changes to the translated files is to update the number and indicate that 336 and 337 are taken somewhere else. Even though the github compare looks a little wonky for these files.

### CODE REVIEW

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 1]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAME_ALREADY_BOUND_WARN_CNTR0338W=CNTR0338W: The {0} interface of the {1} bean in the {2} module of the {3} application cannot be bound to the {4} name location. The {5} interface of the {6} bean in the {7} module of the {8} application is already bound to the {4} name location. The {4} name location is not accessible.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.explanation=Each enterprise bean interface must specify a unique binding location.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.useraction=Correct the binding file for one of these two bean interfaces so that each interface has a unique binding location.
 
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "Namespace", or "ejb-jar-bnd.xml".
+NAMESPACE_IN_JNDI_BINDING_NAME_CNTR0339E=CNTR0339E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or home interface in the {1} module contains a Namespace.
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.explanation=JNDI binding names must not contain a Namespace which is denoted by the ':' character, the entire string before and including the ':' character is considered a Namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: **suggested edit**
JNDI binding names must not contain a namespace, which is denoted by the ':' character. The entire string before and including the ':' character is considered a namespace.

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 2]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the Namespace.
+
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "Namespace", "ejb-jar-bnd.xml", or "ejblocal:".
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E=CNTR0340E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or local home interface in the {1} module contains an invalid Namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: replace Namespace with namespace

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 3]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+#-------------------------------------------------------------------------------
 
+#-------------------------------------------------------------------------------
+NAME_ALREADY_BOUND_WARN_CNTR0338W=CNTR0338W: The {0} interface of the {1} bean in the {2} module of the {3} application cannot be bound to the {4} name location. The {5} interface of the {6} bean in the {7} module of the {8} application is already bound to the {4} name location. The {4} name location is not accessible.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.explanation=Each enterprise bean interface must specify a unique binding location.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.useraction=Correct the binding file for one of these two bean interfaces so that each interface has a unique binding location.
 
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "Namespace", or "ejb-jar-bnd.xml".
+NAMESPACE_IN_JNDI_BINDING_NAME_CNTR0339E=CNTR0339E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or home interface in the {1} module contains a Namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: for all the following messages: don't capitalize namespace unless it starts the sentence

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 4]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.explanation=JNDI binding names must not contain a Namespace which is denoted by the ':' character, the entire string before and including the ':' character is considered a Namespace.
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the Namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: replace Namespace with namespace

**Comment by olendvcook** [comment]
@dmuelle I've made the suggested changes, thanks!

**Comment by olendvcook** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 5]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAME_ALREADY_BOUND_WARN_CNTR0338W=CNTR0338W: The {0} interface of the {1} bean in the {2} module of the {3} application cannot be bound to the {4} name location. The {5} interface of the {6} bean in the {7} module of the {8} application is already bound to the {4} name location. The {4} name location is not accessible.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.explanation=Each enterprise bean interface must specify a unique binding location.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.useraction=Correct the binding file for one of these two bean interfaces so that each interface has a unique binding location.
 
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "namespace", or "ejb-jar-bnd.xml".
+NAMESPACE_IN_JNDI_BINDING_NAME_CNTR0339E=CNTR0339E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or home interface in the {1} module contains a namespace.
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.explanation=JNDI binding names must not contain a namespace which is denoted by the ':' character. The entire string before and including the ':' character is considered a namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: still missing a comma before "which" clause- should be

JNDI binding names must not contain a namespace, which is denoted by the ':' character.

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 6]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the namespace.
+
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "namespace", "ejb-jar-bnd.xml", or "ejblocal:".
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E=CNTR0340E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or local home interface in the {1} module contains an invalid namespace.
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E.explanation=The local binding name must not contain any other namespace or multiple namespaces besides starting with ejblocal:. Namespaces are denoted by the ':' character, the entire string before and including the ':' character is considered a namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: sorry- missed this on the first pass

The local binding name must not contain any other namespace or multiple namespaces besides starting with ejblocal:.

what does this mean? are either of the following correct?

The local binding name must not contain multiple namespaces or any namespace that does not start with `ejblocal:`

The local binding name can only contain one namespace and it must start with `ejblocal:`

Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: Break into two sentences

Namespaces are denoted by the ':' character.  The entire string before and including the ':' character is considered a namespace.

Comment (olendvcook) [reply dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: @dmuelle How about

`The local binding name can only contain the one 'ejblocal:' namespace`?

**Comment by olendvcook** [comment]
@dmuelle I've made the new suggested changes, thanks!

**Comment by olendvcook** [comment]
#build

**Comment by olendvcook** [comment]
#run-libby-bot

**Comment by olendvcook** [comment]
#build

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 7]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the namespace.
+
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "namespace", "ejb-jar-bnd.xml", or "ejblocal:".
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E=CNTR0340E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or local home interface in the {1} module contains an invalid namespace.
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E.explanation=The local binding name can only contain one namespace and it must start with ejblocal:. Namespaces are denoted by the ':' character. The entire string before and including the ':' character is considered a namespace.
+NAMESPACE_IN_LOCAL_JNDI_BINDING_NAME_CNTR0340E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the invalid namespace.
```
Comment (dmuelle) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: **change**
Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the invalid namespace.
**to**
Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the namespace that is not valid.

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 8]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+#
+#-------------------------------------------------------------------------------
 
+#-------------------------------------------------------------------------------
+NAME_ALREADY_BOUND_WARN_CNTR0338W=CNTR0338W: The {0} interface of the {1} bean in the {2} module of the {3} application cannot be bound to the {4} name location. The {5} interface of the {6} bean in the {7} module of the {8} application is already bound to the {4} name location. The {4} name location is not accessible.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.explanation=Each enterprise bean interface must specify a unique binding location.
+NAME_ALREADY_BOUND_WARN_CNTR0338W.useraction=Correct the binding file for one of these two bean interfaces so that each interface has a unique binding location.
 
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "namespace", or "ejb-jar-bnd.xml".
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: ejb-jar-bnd.xml -> ibm-ejb-jar-bnd.xml

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 9]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+NAMESPACE_IN_JNDI_BINDING_NAME_CNTR0339E=CNTR0339E: The {2} Java Naming and Directory Interface (JNDI) binding name for the {0} bean or home interface in the {1} module contains a namespace.
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.explanation=JNDI binding names must not contain a namespace, which is denoted by the ':' character. The entire string before and including the ':' character is considered a namespace.
+NAMESPACE_IN_BINDING_NAME_CNTR0339E.useraction=Correct the ejb-jar-bnd.xml bindings file to give this bean a valid JNDI binding name by removing the namespace.
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: ejb-jar-bnd.xml -> ibm-ejb-jar-bnd.xml

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops [hunk 10]
```diff
@@ -1539,11 +1539,28 @@ ENTITY_NOT_SUPPORTED_CNTR0335W.explanation=Entity enterprise beans declared in a
...
+
+#-------------------------------------------------------------------------------
+# Do NOT translate "JNDI", "namespace", "ejb-jar-bnd.xml", or "ejblocal:".
```
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: ejb-jar-bnd.xml -> ibm-ejb-jar-bnd.xml

File: dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops (hunk 7 above)
Comment (tkburroughs) [inline dev/com.ibm.ws.ejbcontainer.core/resources/com/ibm/ejs/container/container.nlsprops]: ejb-jar-bnd.xml -> ibm-ejb-jar-bnd.xml

**Comment by olendvcook** [comment]
#run-libby-bot

**Comment by olendvcook** [comment]
@dmuelle @tkburroughs I have made the requested changes, thanks!

**Comment by olendvcook** [comment]
#build

**Comment by tkburroughs** [comment]
@olendvcook Approved

**Comment by olendvcook** [comment]
#build

**Comment by olendvcook** [comment]
#build

**Comment by andrea-pichler** [comment]
No message review requested from L2

**Comment by olendvcook** [comment]
build has 1 failure which is a known ui failure with work item already opened.

#run-libby-bot

**Pull request merged**